core/src/Kokkos_Macros.hpp

```diff
@@ -427,6 +427,11 @@
      defined(__x86_64__) || defined(__PPC64__))
 #define KOKKOS_ENABLE_ASM 1
 #endif
+
+#if KOKKOS_COMPILER_GNU < 500
+#define KOKKOS_IMPL_YOLO_ASSUME_TRIVIALLY_COPYABLE_TO_WORK_AROUND_BUG 1
```

Comment:
Okay, we need to have a discussion about this.

All opinions are valid, but I will happily die on the hill of insisting we allow this in, and am extremely patient on silly points.